connector/Dockerfile

```
@@ -40,4 +41,9 @@ COPY ./connector/.env /app/.env
 
 # Replaces ENV Var statements so they don't overwrite existing ENV Vars
 RUN sed -ri 's/^\s*(\S+)=(.*)$/\1=${\1:-"\2"}/' .env
-ENTRYPOINT bash -c 'set -a && source /app/.env && set +a && exec java -Djava.util.logging.config.file=/app/logging.properties $JVM_ARGS -jar app.jar'
+
+# Change to non-root privilege
+RUN addgroup -S nonroot && adduser -S edc -G nonroot
```

```suggestion
RUN adduser -D -H -s /sbin/nologin edc
```

`adduser` implicitly creates a new group of the same name when not creating a system user with `-S`. `-D` locks the password of the newly created user, `-H` disables home directory creation, and setting `/sbin/nologin` as the shell disables all logins.

---

Also, for image layering purposes, I would move this command further up, ideally to the first command of the stage, as it is unlikely to ever change.

---

Nice, included this!

Some commands need to be ececuted before the new user is used, but I moved it up as far as possible.

---

My idea was to split the calls to `adduser` and the subsequent `USER` command. `adduser` can be the first command of the stage, while the `USER` command is moved down until after the `RUN` commands. It's fine like this as well. Not like creating a user is a majorly expensive operation that requires caching.

---

I would've provided a suggestion, but GitHub unfortunately doesn't let you suggest on lines it deems outside of the "context".